### DISCUSSION TITLE
API Ignoring console=off command line or via payload

### DISCUSSION THREAD

**Discussion by noaheverett** [cloud-hypervisor/cloud-hypervisor#4831] [Q&A] [answered]
Hello! I'm trying to set the boot console to serial=tty and console=off which shows full boot info via the API. I've been able to get this to work by using the command line to start a VM
```
cloud-hypervisor-static \
--kernel ./hypervisor-fw \
--disk path=focal-server-cloudimg-amd64.raw path=/tmp/ubuntu-cloudinit.img \
--cpus boot=4 \
--memory size=1024M \
--net "tap=,mac=,ip=,mask=" \
--serial tty \
--console off
```
However if I start the API server via:
`cloud-hypervisor-static --api-socket /tmp/cloud-hypervisor.sock --console off --serial tty`

It always starts with the normal console when starting a VM via:
```
curl --unix-socket /tmp/cloud-hypervisor.sock -i \
    -X PUT 'http://localhost/api/v1/vm.create'  \
    -H 'Accept: application/json'               \
    -H 'Content-Type: application/json'         \
    -d '{
        "cpus":{"boot_vcpus": 1, "max_vcpus": 1},
        "payload":{"kernel":"/home/ark/chtest/hypervisor-fw", "cmdline":"console=off serial=tty root=/dev/vda1 rw"},
        "disks":[{"path":"/home/ark/chtest/focal-server-cloudimg-amd64.raw"}],
        "rng":{"src":"/dev/urandom"},
        "net":[{"tap":"tap0", "ip":"172.20.0.42", "mask":"255.255.255.0", "mac":"12:34:56:78:90:01"}]
        }'
```
I've tried variations of the above command where `cmdline` param removes console/serial all together or setting them to different variations and seems to start in the normal console which doesn't show boot messages. I've also tried starting the API server without the `--console/--serial` and setting it via the API create call as well.

Thanks for your help!

**Comment by rbradford** [comment, marked as answer]
If you want to the equivalent of `--serial tty --console off` then you need to extend your JSON definition as console is on by default:
```json
    "serial": {
      "mode": "Tty",
    },
    "console": {
      "mode": "Off",
    },
```

**Reply by noaheverett** [reply]
Ah yes beautiful, it works, thank you! Combined with API documentation at #4828 this is a huge benefit. Cheers!
